src/test/regress/sql/with.sql

```diff
@@ -1187,3 +1187,19 @@ create temp table with_test (i int);
 with with_test as (select 42) insert into with_test select * from with_test;
 select * from with_test;
 drop table with_test;
+
+-- Make sure we create a correct tuple descriptor for SubqueryScan above
+-- ShareInputScan with ModifyTable below.
+CREATE TABLE with_test (i int) DISTRIBUTED BY (i);
```

Comment:
What about using the with_test table from the previous test instead of recreating?